Libraries/Blob/URL.js

```diff
@@ -128,7 +128,7 @@ export class URL {
     let baseUrl = null;
     if (!base || validateBaseUrl(url)) {
       this._url = url;
-      if (!this._url.endsWith('/')) {
+      if (!this._url.endsWith('/') && url.indexOf('?') < 0) {
```

Comment:
Instead, could you search for `?` at the top and split out `queryString` from `url` so that most of this logic can stay intact and not worry about query string until below where we attempt to parse it?

Reply:
@yungsters This instance of the check doesn't pertain to the parsing of the query string value. It simply checks to see if it's safe to append a trailing `/` to the passed in URL. Unfortunately moving the check to the top would make parsing logic more complex later since we'd have to account for both use cases of `url` (either the full url or a url fragment).